From: Takashi Iwai <tiwai@suse.de>
Subject: kABI workaround for HD-audio generic parser
Patch-mainline: Never, kABI workaround
References: git-fixes

The recent fix
  patches.suse/ALSA-hda-generic-Add-option-to-enforce-preferred_dac.patch
introduced a new bit field in hda_gen_spec struct.  As a standard kABI
workaround, move it to the tail of other bit fields and protect iv via
__GENKSYMS__.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 sound/pci/hda/hda_generic.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/sound/pci/hda/hda_generic.h
+++ b/sound/pci/hda/hda_generic.h
@@ -240,7 +240,6 @@ struct hda_gen_spec {
 	unsigned int power_down_unused:1; /* power down unused widgets */
 	unsigned int dac_min_mute:1; /* minimal = mute for DACs */
 	unsigned int suppress_vmaster:1; /* don't create vmaster kctls */
-	unsigned int obey_preferred_dacs:1; /* obey preferred_dacs assignment */
 
 	/* other internal flags */
 	unsigned int no_analog:1; /* digital I/O only */
@@ -252,6 +251,11 @@ struct hda_gen_spec {
 	unsigned int skip_verbs:1; /* don't apply verbs at snd_hda_gen_init() */
 #endif
 
+#ifndef __GENKSYMS__
+	/* newly added flags: moved here due to kABI compatibility */
+	unsigned int obey_preferred_dacs:1; /* obey preferred_dacs assignment */
+#endif
+
 	/* additional mute flags (only effective with auto_mute_via_amp=1) */
 	u64 mute_bits;
 
